shims/common/src/main/scala/io/glutenproject/GlutenConfig.scala

```diff
@@ -186,10 +186,18 @@ class GlutenConfig(conf: SQLConf) extends Logging {
     }
   }
 
+  def memoryIsolation: Boolean = conf.getConf(COLUMNAR_MEMORY_ISOLATION)
+
   def offHeapMemorySize: Long = conf.getConf(COLUMNAR_OFFHEAP_SIZE_IN_BYTES)
 
   def taskOffHeapMemorySize: Long = conf.getConf(COLUMNAR_TASK_OFFHEAP_SIZE_IN_BYTES)
 
+  def conservativeOffHeapMemorySize: Long =
+    conf.getConf(COLUMNAR_CONSERVATIVE_OFFHEAP_SIZE_IN_BYTES)
+
+  def conservativeTaskOffHeapMemorySize: Long =
+    conf.getConf(COLUMNAR_CONSERVATIVE_TASK_OFFHEAP_SIZE_IN_BYTES)
+
```

Comment:
Some questions here:
1. what's the difference for these two configs? 
2. what does `conservative` mean?
3. seems `conservativeOffHeapMemorySize` was not used?

Reply:
"Conservative" means the max size Gluten can consider is "safe" to use.  The two new options are set in `GlutenPlugin.scala` using the following code:

```java
// Pessimistic off-heap sizes, with the assumption that all non-borrowable storage memory
// determined by spark.memory.storageFraction was used.
val fraction = 1.0d - conf.getDouble("spark.memory.storageFraction", 0.5d)
val conservativeOffHeapSize = (offHeapSize
  * fraction).toLong
conf.set(
  GlutenConfig.GLUTEN_CONSERVATIVE_OFFHEAP_SIZE_IN_BYTES_KEY,
  conservativeOffHeapSize.toString)
val conservativeOffHeapPerTask = conservativeOffHeapSize / taskSlots
conf.set(
  GlutenConfig.GLUTEN_CONSERVATIVE_TASK_OFFHEAP_SIZE_IN_BYTES_KEY,
  conservativeOffHeapPerTask.toString)
```

The difference between the options and "non-conservative" options is that the "conservative" ones take storage memory into account. Assuming Spark had used 30% off-heap memory in storage memory pool, the memory would not be evict although a "borrow" is request from execution memory pool.

I think for stability, we may need to by default use "conservative" options since it's safter in most cases. I left the "non-conservative" options unchanged because of compatibility consideration.

Reply:
Besides, I don't want users to set these "auto-generated" options. But we don't develop a general way to guard against that yet.

Reply:
> seems conservativeOffHeapMemorySize was not used?

Yes. But it's worth to keep it for future use.

Reply:
Thanks for detailed explanation! I have realized the "conservative" meaning.
 
> we may need to by default use "conservative" options since it's safter in most cases.

Please keep a switch config for that, Spark already has `maybeGrowExecutionPool` to shrink storagePool and executionPool, I prefer let Spark control this logic. If use "conservative" option by default, we may not fully utilize the unused storage memory.

Reply:
I meant to use "conservative" options for just Gluten's related codes that requires reading the off-heap size. For example shuffle writer and partial agg. It's not a goal to touch vanilla Spark's memory management.

Reply:
Worth noting that Spark's "storage region size" determined by `spark.memory.storageFraction` is not evictable if in use. That's why we add the conservative options.